algorithms/test/benchmark/box_search.py

```diff
@@ -0,0 +1,84 @@
+from dist_sqr import dist_sqr
+
+
+class BoxSearch:
+    def __init__(self, search_radius: float, epsilon: float, field_size: float):
+        """
+
+        :param search_radius: minimal distance for particles to become neighbors
+        :param epsilon: distance calculation precision
+        :param field_size: working field upper bounder
+        """
+        self.search_radius = search_radius
+        self.epsilon = epsilon
+        self.field_size = field_size
+        self._boxes_in_row = int(self.field_size / self.search_radius)
+        self._total_boxes = self._boxes_in_row ** 2
+        self._box_neighbors = self._find_neighbor_boxes()
+
+    def _find_points_boxes(self, points: list[list]) -> list[list]:
+        box_indexes = [[] for box in range(self._total_boxes)]
+
+        for point_index, point in enumerate(points):
+            col = round(point[0] / self.search_radius)
+            row = round(point[1] / self.search_radius)
+            box_id = int(row * self._boxes_in_row + col)
+            box_indexes[box_id].append(point_index)
+
+        return box_indexes
+
+    def _find_one_box_neighbors(self, box_id: int) -> list[int]:
+        col = box_id % self._boxes_in_row
+        row = round(box_id / self._boxes_in_row)
+        possible_neighbors = [
+            (row, col + 1),
+            (row, col - 1),
+            (row + 1, col),
+            (row - 1, col),
+            (row + 1, col + 1),
+            (row - 1, col + 1),
+            (row + 1, col - 1),
+            (row - 1, col - 1),
+        ]
+        box_neighbors_ids = [row * self._boxes_in_row + col for row, col in possible_neighbors
+                             if (0 <= col < self._boxes_in_row) and (0 <= row < self._boxes_in_row)]
+
+        return box_neighbors_ids
+
+    def _find_neighbor_boxes(self) -> list[list]:
```

Comment:
please put this method after init. it would be easier to read code if the methods are ordered by the time they called.

Reply:
done